### PR TITLE
configurable lease check wait for non-renewable secrets

### PR DESCRIPTION
**Pull request by rgeoghegan**
This should be a fix for https://github.com/hashicorp/consul-template/issues/1526

Now the vault config has a `lease_renewal_threshold` parameter which controls the
fraction of how long of the original lease duration consul-template should wait to
ask for a new secret on non-renewable secrets (like for PKIs). By default, consul-template
will wait (90 +/- 5)% of the lease time.

### CODE REVIEW

File: config/vault.go [hunk 1]
```diff
@@ -310,8 +328,9 @@ func (c *VaultConfig) GoString() string {
 		"Token:%t, "+
 		"VaultAgentTokenFile:%t, "+
 		"Transport:%#v, "+
-		"UnwrapToken:%s"+
+		"UnwrapToken:%s, "+
```
Comment (eikenb) [inline config/vault.go:331]: Drive-by formatting fix. Thanks!

**Review by eikenb — APPROVED** [review]
Seems like you've covered all the bases.
LGTM.

File: config/vault.go (hunk 1 above)
Comment (rgeoghegan) [reply config/vault.go:331]: I was trying to figure out why the unit tests would not pass, and I noticed this typo when my print statements were all weird.
